kmatch/kmatch.py

```diff
@@ -152,3 +152,30 @@ def match(self, value):
                 suppress_key_errors class variable is False
         """
         return self._match(self._compiled_pattern, value)
+
+    def get_field_keys(self, pattern=None):
+        """
+        Builds a set of all field keys used in the pattern including nested fields.
+
+        :param pattern: The kmatch pattern to get field keys from or None to use self.pattern
+        :type pattern: list or None
+        :returns: A set object of all field keys used in the pattern
+        :rtype: set
+        """
+        # Use own pattern or passed in argument for recursion
+        pattern = pattern or self.pattern
+        keys = set()
+
+        # Valid pattern length can only be 2 or 3
+        if len(pattern) == 2:
+            if pattern[0] in ('&', '|', '^'):
+                # Pass each nested pattern to get_field_keys
+                for filter_item in pattern[1]:
+                    keys = keys.union(self.get_field_keys(filter_item))
+            else:
+                # pattern[0] == '!'
+                keys = keys.union(self.get_field_keys(pattern[1]))
+        else:
+            # Pattern length is 3
+            keys.add(pattern[1])
```

Comment:
Ok, so I didn't pay enough attention before, but it looks like if someone passes in a pattern that only contains one element, this will error out in a non-obvious way. Any chance you could add an explanation of this in the docs, handle the `len(pattern) == 3` case explicitly, and raise an error if the length is invalid?